core/domain/calculation_registry_test.py

```diff
@@ -27,6 +27,11 @@ class CalculationRegistryTests(test_utils.GenericTestBase):
     """Provides testing of the calculation registry."""
 
     def test_get_calculation_by_id(self) -> None:
+        self.assertTrue(
+            isinstance(
+                calculation_registry.Registry.get_calculation_by_id(
+                    'AnswerFrequencies'),
+                models.AnswerFrequencies))
```

Comment:
I'm a bit confused about how these lines increase coverage. It appears to be the same as Lines 35-39. Could you please explain?

Reply:
Hi @kevintab95,
Here if calculation_id is not in cls._calculations_dic
https://github.com/oppia/oppia/blob/b81a1365c5dcd7999c142abd3a0374eb8bf83717/core/domain/calculation_registry.py#L101
then
https://github.com/oppia/oppia/blob/b81a1365c5dcd7999c142abd3a0374eb8bf83717/core/domain/calculation_registry.py#L102
This will add that calculation id to cls._calculations_dic
now if i write https://github.com/oppia/oppia/blob/b81a1365c5dcd7999c142abd3a0374eb8bf83717/core/domain/calculation_registry_test.py#L30
two times then,
for the first time AnswerFrequencies is not in cls._calculations_dic
so AnswerFrequencies  inserted
but when the second time called 
AnswerFrequencies already present.
so it jumps from 101->103
that is the missing coverage.
Summary:
I have to write a case such that calculation_id is prsent in cls._calculations_dic, but initialy its empty.
so first I added calculation_id and then i checked the same calculation_id .

Reply:
Hi @kevintab95 @gp201 PTAL!.

Reply:
Hm, can we have a separate test case to explicitly cover this case and add an explanatory comment? Right now, it looks like this is covered as a "side effect", it may not be clear to a future developer why the statements are duplicated.